3_javascript/javascript3/week3/screenshot_app/script.js

```diff
@@ -0,0 +1,151 @@
+          "X-RapidAPI-Host": "website-screenshot6.p.rapidapi.com",
+        },
+      };
+      fetch(urlAPI, options)
+        .then((data) => data.json())
+        .then((screenshot) => {
+          if (screenshot.screenshotUrl) {
+            saveScreenshot(screenshot);
+          } else {
+            alert("The URL is not valid.");
+          }
+        })
+        .catch((error) => console.error("Error", error))
+        .finally(() => {
+            loadingIndicator.style.display = "none";
+          });
+    });
+  }
+
+  function getScreenshotURL(URL) {
+    return URL;
+  }
+
+  function saveScreenshot(screenshot) {
+    fetch(
+      "https://crudcrud.com/api/5f0deea212a747f181a722a6a4806a42/screenshot",
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(screenshot),
+      }
+    )
+      .then(() => {
+        loadScreenshots();
+      })
+      .catch((error) => console.error("Error", error));
+  }
+
+  function loadScreenshots() {
+    fetchScreenshots()
+      .then((data) => {
+        const screenshotList = document.getElementById("screenshotList");
+        screenshotList.innerHTML = "";
+        selectScreenshot.innerHTML = "";
+
+        data.forEach((screenshot, index) => {
+          const linkElement = createScreenshotLink(screenshot, index);
+          const listItem = createScreenshotListItem(linkElement);
+          screenshotList.appendChild(listItem);
+
+          const optionElement = createScreenshotOption(index);
+          selectScreenshot.appendChild(optionElement);
+        });
+      })
+      .catch((error) => console.error("Error", error));
+  }
+
+  async function fetchScreenshots() {
+    return fetch(
+      "https://crudcrud.com/api/5f0deea212a747f181a722a6a4806a42/screenshot",
+      {
+        method: "GET",
+      }
+    ).then((response) => response.json());
+  }
+
+  function createScreenshotLink(screenshot, index) {
+    const linkElement = document.createElement("a");
+    linkElement.href = screenshot.screenshotUrl;
+    linkElement.textContent = `Screenshot ${index + 1}`;
+    linkElement.target = "_blank";
+    return linkElement;
+  }
+
+  function createScreenshotListItem(linkElement) {
+    const listItem = document.createElement("li");
+    listItem.appendChild(linkElement);
+    return listItem;
+  }
+
+  function createScreenshotOption(index) {
+    const optionElement = document.createElement("option");
+    optionElement.value = index + 1;
+    optionElement.text = `Screenshot ${index + 1}`;
+    return optionElement;
+  }
+
+  function deleteSelectedScreenshot() {
+    const selectedScreenshotNumber = parseInt(selectScreenshot.value);
+
+    if (!isNaN(selectedScreenshotNumber)) {
+      const selectedIndex = selectedScreenshotNumber - 1;
+
+      fetchScreenshots()
+        .then((data) => {
+          if (selectedIndex >= 0 && selectedIndex < data.length) {
```

Comment:
There is an easier way to check if the index is available in the array:
```js
if (data[selectedIndex] !== undefined) {
  // ....
}
```

Reply:
Nice!